protocols/kad/src/behaviour.rs

```diff
@@ -1978,21 +1980,66 @@ impl<TStore> NetworkBehaviour for Kademlia<TStore>
 where
     TStore: RecordStore + Send + 'static,
 {
-    type ConnectionHandler = KademliaHandlerProto<QueryId>;
+    type ConnectionHandler = KademliaHandler<QueryId>;
     type OutEvent = KademliaEvent;
 
-    fn new_handler(&mut self) -> Self::ConnectionHandler {
-        KademliaHandlerProto::new(KademliaHandlerConfig {
-            protocol_config: self.protocol_config.clone(),
-            allow_listening: true,
```

Comment:
Uuf, when did we miss this?

Reply:
What do you mean?

Reply:
If I understand correctly `allow_listening` is always `true`, thus a dummy configuration option, thus fine to remove. I am assuming that this is a relict from a past refactoring. Not directly related to this pull request. Resolving the comment.

Reply:
It is not removed in this PR but I guess we should remove it, I'll open an issue!